### DISCUSSION TITLE
how to obtain retry info in processor?

### DISCUSSION THREAD

**Discussion by schester44** [timgit/pg-boss#442] [Q&A]
I have a job that has a retryLimit set. I want to call some other code when the retryLimit has been reached.

With bullmq, i'm able to do something like `const isLastAttempt = job.attemptsStarted === job.opts.attempts;`

Does pg-boss have something similar? Struggling to find what i'm looking for in the docs.

**Comment by schester44** [comment]
looks like i can make use of the getJobById function to get retry info

**Comment by timgit** [comment]
Try using `includeMetadata` option in `fetch()`:
https://github.com/timgit/pg-boss/tree/master/docs#fetchname-options
